src/components/SearchModal/CommonBases.tsx

```diff
@@ -47,19 +53,39 @@ export default function CommonBases({
       <AutoRow gap="4px">
         {bases.map((currency: Currency) => {
           const isSelected = selectedCurrency?.equals(currency)
+          const token_address = currency instanceof Token ? currency?.address : undefined
```

Comment:
i wonder if similar to having `Props` for components we should create a `formatEventProperties` helper function for each component - just to keep. component logic clean?

Reply:
good call, added helper function.